sklearn/feature_selection/from_model.py

```diff
@@ -199,9 +215,33 @@ def _get_support_mask(self):
                 'Either fit the model before transform or set "prefit=True"'
                 ' while passing the fitted estimator to the constructor.')
         scores = _get_feature_importances(estimator)
-        self.threshold_ = _calculate_threshold(estimator, scores,
-                                               self.threshold)
-        return scores >= self.threshold_
+        mask = np.zeros(scores.shape, dtype=bool)
+        if self.threshold is not None and self.max_n_features is not None:
+            self.threshold_ = _calculate_threshold(estimator, scores,
+                                                   self.threshold)
+            mask = scores >= self.threshold_
+            if self.max_n_features == 'all':
+                pass
+            elif self.max_n_features == 0:
+                mask[:] = False
+            else:
+                scores = np.ma.masked_array(scores, mask=1-mask)
```

Comment:
I need to find top n scores from `scores` where the `mask` is `1`. Lets say, `scores = [7, 1, 4, 2, 3]`, `self.threshold_ = 3` and `self.max_n_features=2`. What is the easiest way to create the output mask? `[1, 0, 1. 0, 0]`. Actually I didn't find an easier/intuitive way.